src/database/adatabase.cpp

```diff
@@ -715,3 +715,61 @@ QMap<QString, QString> ADatabase::databaseSummary(const QString &db_path)
 
     return return_values;
 }
+
+/*!
+ * \brief ADatabase::createBackup copies the currently used database to an external backup location provided by the user
+ * \param dest_file This is the full path and filename of where the backup will be created, e.g. 'home/Sully/myBackups/backupFromOpl.db'
+ */
+bool ADatabase::createBackup(const QString& dest_file)
+{
+    INFO << "Backing up current database to: " << dest_file;
+    ADatabase::disconnect();
+    QFile file(databaseFile.absoluteFilePath());
+    DEB << "File:" << file;  // [G]: Check adebug.h got INFO WARN, ... additions and discuss convention of use.
+
+    if (!file.copy(dest_file)) {
```

Comment:
I think this function also needs to be able to overwrite (delete, copy) existing files. QFileDialog asks for confirmation if the user selects an existing file as output, so we could implement the same logic as is used in `restoreBackup`

Reply:
I missed this i think.
So the function should by default overwrite the dest_file?
The assumption is that QFileDialog is our "guard" because it prompts the overwrite?
That means that the user pressing yes will overwrite an existing file where as no means he _will_ put a new file which doesnt exist and just create it
Please confirm my understanding, it kind of flew over my head

Reply:
The idea was that QFileDialog prompts the user to confirm if he really wants to select the existing file, which is default behaviour as far as I'm aware. If that's the case, then the function should silently overwrite since confirmation has already been given. But we would of course have to make sure that it is only called if a confirmation has been given. I need to look into what QFileDialog returns when the action is cancelled, let me get back to you on that.